### PR TITLE
fix static linking when using Intel OneAPI mkl

### PR DESCRIPTION
**Pull request by he-b**
solves #36988

### CODE REVIEW

**Review by rscohn2 — APPROVED** [review]
Thanks for fixing mkl static linking issue.

**Comment by rscohn2** [comment]
Since this has been sitting for a while, I rebased.

@xiaoye @gchavez2 @balay @pghysels @liuyangzhuan do you have objections? ld_flags is used for other libs as well.

File: var/spack/repos/builtin/packages/superlu-dist/package.py [hunk 1]
```diff
@@ -89,8 +89,8 @@ def append_from_variant(*args):
         append_define("CMAKE_CXX_COMPILER", spec["mpi"].mpicxx)
         append_define("CMAKE_INSTALL_LIBDIR", self.prefix.lib)
         append_define("CMAKE_INSTALL_BINDIR", self.prefix.bin)
-        append_define("TPL_BLAS_LIBRARIES", spec["blas"].libs)
-        append_define("TPL_LAPACK_LIBRARIES", spec["lapack"].libs)
+        append_define("TPL_BLAS_LIBRARIES", spec["blas"].libs.ld_flags)
+        append_define("TPL_LAPACK_LIBRARIES", spec["lapack"].libs.ld_flags)
```
Comment (balay) [inline var/spack/repos/builtin/packages/superlu-dist/package.py:93]:
```
balay@p1 /home/balay/git-repo/github/spack (develop =)
$ git grep TPL_BLAS_LIBRARIES
var/spack/repos/builtin/packages/butterflypack/package.py:            "-DTPL_BLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
var/spack/repos/builtin/packages/gptune/package.py:            "-DTPL_BLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
var/spack/repos/builtin/packages/strumpack/package.py:            "-DTPL_BLAS_LIBRARIES=%s" % spec["blas"].libs.joined(";"),
var/spack/repos/builtin/packages/superlu-dist/package.py:        append_define("TPL_BLAS_LIBRARIES", spec["blas"].libs)
```

Perhaps using `libs.joined(";")` is the right thing to do?

Comment (balay) [reply var/spack/repos/builtin/packages/superlu-dist/package.py:93]: But yeah - I do see `libs.ld_flags` usage - so perhaps this is ok. Will defer this for @lingnanyuan @pghysels 
```
            "TPL_PARMETIS_LIBRARIES", [spec["parmetis"].libs.ld_flags, spec["metis"].libs.ld_flags]
```

Comment (rscohn2) [reply var/spack/repos/builtin/packages/superlu-dist/package.py:93]: MKL static requires some extra switches that are not libraries:
https://github.com/spack/spack/blob/a9538f2c2985558ed4863bcd3f64ac3b1db4cbde/lib/spack/spack/build_systems/oneapi.py#L203

**Comment by rscohn2** [comment]
Is the change ok?

**Comment by liuyangzhuan** [comment]
> Is the change ok?

Sorry I missed this request. Yes, it looks no issue to me.

**Review by liuyangzhuan — APPROVED** [review]

**Comment by Marshdweller** [comment]
Hi everyone. I have the same issue. Just curious if there is progress here.

**Comment by rscohn2** [comment]
> Hi everyone. I have the same issue. Just curious if there is progress here.

Thanks for the reminder. I think the review comments were resolved and I merged it.

**Pull request merged**